### PR TITLE
Element with attr selector

### PR DESCRIPTION
**Pull request by michalwski**
New functions `exml_query:subelement_with_attr/3,4` - returns first child with given attribute and value.

This PR refers to #27 

TODO:
* [x] use this function in `exml_query:path`
* [x] implement `exml_query:subelements_wit_attr` which returns all subelements with given attribute and value
* [x] use the abce in `exml_query:paths`

### CODE REVIEW

File: src/exml_query.erl [hunk 1]
```diff
@@ -20,13 +21,16 @@
...
 -type element_with_name_and_ns () :: {element_with_ns, binary(), binary()}.
+-type element_with_attr_of_value () :: {element_with_attr, binary(), binary()}.
 
 -type path() :: [cdata | %% selects cdata from the element
                  {element, binary()} | % selects subelement with given name
                  {attr, binary()} | % selects attr of given name
                  element_with_ns() | % selects subelement with given namespace
-                 element_with_name_and_ns() % selects subelement with given name and namespace
+                 element_with_name_and_ns() | % selects subelement with given name and namespace
+                 element_with_attr_of_value() % selects subelement with given attritube and value
```
Comment (arkgil) [inline src/exml_query.erl]: `attribute` instead of `attritube` 😄

File: test/exml_query_tests.erl [hunk 2]
```diff
@@ -61,6 +61,21 @@ no_element_with_ns_query_test() ->
                  exml_query:subelement_with_ns(chat_marker(),
                                                <<"wrong">>, none)).
 
+element_with_attr_query_test() ->
+    ?assertEqual(xml(<<"<received xmlns='urn:xmpp:chat-markers:0'
+                                id='0047ee62-9418-4ef8-abd8-0d08e4140b72'/>)">>),
+                 exml_query:subelement_with_attr(chat_marker(), <<"xmlns">>,
+                                                 <<"urn:xmpp:chat-markers:0">>)),
+
+    ?assertEqual(xml(<<"<received xmlns='urn:xmpp:chat-markers:0'
```
Comment (arcusfelis) [inline test/exml_query_tests.erl]: We should also test the case when there are more than one sub-element in the message.
The test only test the happy-way scenario.
I.e. if we modify element_with_attr to just return first child, the tests would not spot the change.

**Comment by michalwski** [comment]
Yes, I did merge instead of rebase. This was on purpose since the ref on this branch is used in customer project and I don't want to loose it.

File: test/exml_query_tests.erl [hunk 3]
```diff
@@ -61,6 +61,21 @@ no_element_with_ns_query_test() ->
...
+                                id='0047ee62-9418-4ef8-abd8-0d08e4140b72'/>)">>),
+                 exml_query:path(chat_marker(), [{element_with_attr, <<"xmlns">>,
+                                                  <<"urn:xmpp:chat-markers:0">>}])).
+
+no_element_with_attr_query_test() ->
```
Comment (michalwski) [inline test/exml_query_tests.erl]: @arcusfelis I think this is the test case you named "nothing matches scenario" do you agree?

Comment (arcusfelis) [reply test/exml_query_tests.erl]: yep

**Pull request merged**